src/nodes/parameter.h

```diff
@@ -48,6 +50,13 @@ class ParameterBase : public Serialisable<>
         Input,     /* Indicates that the parameter is meant to be a sink for data and not a source */
         Output,    /* Indicates that the parameter is meant to be a source of data and not a sink */
     };
+    // Allowed Edge Connections
+    enum EdgeLimit
+    {
+        None, /* No edges are allowed */
+        One,  /* Exactly one edge is allowed */
+        Many  /* Many edges are allowed */
+    };
```

Comment:
`EdgeLimit::None` is feels uncomfortably close to "there are no limits on this edge!" - if I'm not mistaken the real meaning is basically the opposite, an edge with no connections. How about the following
```suggestion
enum class EdgePopulation
{
   None, /* No edges are allowed */
   One, /* Exactly one edge is allowed */
   Many /* Many edges are allowed */
}
```
If "Many" is too imprecise to constitute a Population, we could also call the enum `ExpectedEdges`, or `HasEdges`